pkg/appfile/parser.go

```diff
@@ -307,53 +319,59 @@ output: {
 }`, targetWokrloadGVK.GroupVersion().String(), targetWokrloadGVK.Kind)
 
 	// re-use the way CUE module generates comp & acComp
-	comp, acComp, err := generateComponentFromCUEModule(c, wl, appName, revision, ns)
+	comp, acComp, needTracker, err := generateComponentFromCUEModule(c, wl, appName, revision, ns)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 
 	release, repo, err := helm.RenderHelmReleaseAndHelmRepo(wl.Helm, wl.Name, appName, ns, wl.Params)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 	rlsBytes, err := json.Marshal(release.Object)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 	repoBytes, err := json.Marshal(repo.Object)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 	comp.Spec.Helm = &v1alpha2.Helm{
 		Release:    runtime.RawExtension{Raw: rlsBytes},
 		Repository: runtime.RawExtension{Raw: repoBytes},
 	}
-	return comp, acComp, nil
+	return comp, acComp, needTracker, nil
 }
 
 // evalWorkloadWithContext evaluate the workload's template to generate component and ACComponent
-func evalWorkloadWithContext(pCtx process.Context, wl *Workload, appName, compName string) (*v1alpha2.Component, *v1alpha2.ApplicationConfigurationComponent, error) {
+func evalWorkloadWithContext(pCtx process.Context, wl *Workload, appName, compName string, appNs string) (*v1alpha2.Component, *v1alpha2.ApplicationConfigurationComponent, bool, error) {
```

Comment:
I think the main info is contained in the returned value `components`, so let's check outside. Don't need to add a return bool here.